app/models/shopping_list_item_builder.rb

```diff
@@ -1,59 +1,45 @@
 # frozen_string_literal: true
 
 class ShoppingListItemBuilder
-  attr_reader :ingredients, :meal_plan, :recipe
+  attr_reader :shopping_list, :ingredients
 
-  def initialize(shopping_list:, single_ingredient: nil, meal_plan:, recipe:)
-    @shopping_list = shopping_list
-    @single_ingredient = single_ingredient
-    @meal_plan = meal_plan
-    @recipe = recipe
-    @ingredients = assign_ingredients
+  def initialize(shopping_list_id:, ingredient_ids:)
+    @shopping_list = ShoppingList.find_by(id: shopping_list_id)
+    @ingredients = Ingredient.where(id: ingredient_ids)
   end
 
+  def add_items_to_list
+    self.ingredients.each do |ingredient|
```

Comment:
I added the unneeded `self` here I and in `add_ingredient_to_list` just to try to be more clear that it's calling the getter method set in the `attr_reader` rather than referencing a local variable.

Reply:
I appreciate that clarity :)